nifi-nar-bundles/nifi-adx-bundle/nifi-adx-processors/src/main/java/org/apache/nifi/processors/adx/QueryAzureDataExplorer.java

```diff
@@ -0,0 +1,193 @@
+        @WritesAttribute(attribute = "ADX_QUERY_ERROR_MESSAGE", description = "Azure Data Explorer error message."),
+        @WritesAttribute(attribute = "ADX_EXECUTED_QUERY", description = "Azure Data Explorer executed query.")
+})
+public class QueryAzureDataExplorer extends AbstractProcessor {
+    public static final String ADX_QUERY_ERROR_MESSAGE = "adx.query.error.message";
+    public static final String ADX_EXECUTED_QUERY = "adx.executed.query";
+    public static final String RELATIONSHIP_SUCCESS = "SUCCESS";
+
+    public static final String RELATIONSHIP_FAILED = "FAILED";
+    public static final String RELATIONSHIP_FAILED_DESC = "Relationship for failure";
+    public static final String RELATIONSHIP_SUCCESS_DESC = "Relationship for success";
+
+    public static final Relationship SUCCESS = new Relationship.Builder()
+            .name(RELATIONSHIP_SUCCESS)
+            .description(RELATIONSHIP_SUCCESS_DESC)
+            .build();
+    public static final Relationship FAILED = new Relationship.Builder()
+            .name(RELATIONSHIP_FAILED)
+            .description(RELATIONSHIP_FAILED_DESC)
+            .build();
+    public static final PropertyDescriptor DB_NAME = new PropertyDescriptor
+            .Builder().name(AzureAdxSourceProcessorParameter.DB_NAME.name())
+            .displayName(AzureAdxSourceProcessorParameter.DB_NAME.getParamDisplayName())
+            .description(AzureAdxSourceProcessorParameter.DB_NAME.getParamDescription())
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor ADX_QUERY = new PropertyDescriptor
+            .Builder().name(AzureAdxSourceProcessorParameter.ADX_QUERY.name())
+            .displayName(AzureAdxSourceProcessorParameter.ADX_QUERY.getParamDisplayName())
+            .description(AzureAdxSourceProcessorParameter.ADX_QUERY.getParamDescription())
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor ADX_SOURCE_SERVICE = new PropertyDescriptor
+            .Builder().name(AzureAdxSourceProcessorParameter.ADX_SOURCE_SERVICE.name())
+            .displayName(AzureAdxSourceProcessorParameter.ADX_SOURCE_SERVICE.getParamDisplayName())
+            .description(AzureAdxSourceProcessorParameter.ADX_SOURCE_SERVICE.getParamDescription())
+            .required(true)
+            .identifiesControllerService(AdxSourceConnectionService.class)
+            .build();
+    private final ObjectMapper objectMapper = new ObjectMapper();
+    private Set<Relationship> relationships;
+    private List<PropertyDescriptor> descriptors;
+    private Client executionClient;
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        this.descriptors = List.of(ADX_SOURCE_SERVICE,DB_NAME,ADX_QUERY);
+        this.relationships = Set.of(SUCCESS,FAILED);
```

Comment:
These collections can be declared statically and the `init` method can be removed.